src/ui/camera.js

```diff
@@ -1588,6 +1647,22 @@ class Camera extends Evented {
             delta > 180 ? -360 :
             delta < -180 ? 360 : 0;
     }
+
+    // emulates frame function for some transform
+    _emulate(frame: Function, options: {duration: number, frameRate?: number, initialTransform?: Transform}) {
+        const frameRate = options.frameRate || 15;
+        const frameTime = 1000 / (frameRate * options.duration);
+        const tr = options.initialTransform || this.transform;
+
+        const transforms = [];
+        const emulateFrame = frame(tr.clone());
+        for (let i = 0; i <= 1; i += frameTime) {
```

Comment:
Nit: due to floating point errors, there's a chance we'll overshoot the last frame — it might be more reliable to increment with integer (`let i = 0; i <= numFrames; i++`) and then `emulate(i / numFrames)`.